skyportal/handlers/api/source.py

```diff
@@ -249,6 +255,10 @@ def get(self, obj_id=None):
             )
             if s is None:
                 return self.error("Invalid source ID.")
+            if "ps1" not in [thumb.type for thumb in s.thumbnails]:
+                IOLoop.current().add_callback(
+                    lambda: add_ps1_thumbnail_and_push_ws_msg(s, self)
```

Comment:
What is the scenario in which this will be triggered?

Reply:
If the PS1 thumbnail doesn't exist yet when a single source is fetched, we kick off the process that generates it